lib/aws/broker/subscriber.rb

```diff
@@ -0,0 +1,82 @@
+require 'aws/broker/base'
+
+module Aws
+  class Broker
+    class Subscriber < Base
+
+      def initialize(topic, queue=nil)
+        @topic = topic
+        @queue = queue || queue_name
+      end
+
+      def subscribe
+        return unless enabled?
+        create_queue
+        create_topic
+        sns.subscribe(
+          topic_arn: @topic_arn,
+          protocol:  'sqs',
+          endpoint:  @queue_arn
+        )
+        set_queue_policy
+      end
+
+    private
+
+      def queue_name
+        if config.queue_prefix
+          "#{config.queue_prefix}-#{@topic}"
+        else
+          @topic
+        end
+      end
+
+      def create_queue
+        @queue_url = sqs.create_queue(
+          queue_name: @queue
+        ).queue_url
+        @queue_arn = sqs.get_queue_attributes(
```

Comment:
want to break this out into another method?

Reply:
sure